bcipy/simulator/task/copy_phrase.py

```diff
@@ -0,0 +1,138 @@
+        super().__init__(win=None,
+                         daq=None,
+                         parameters=parameters,
+                         file_save=file_save,
+                         signal_models=signal_models,
+                         language_model=language_model,
+                         fake=False)
+        self.save_session_every_inquiry = False
+        self.samplers = samplers
+
+    def init_evidence_evaluators(
+            self, signal_models: List[SignalModel]) -> List[EvidenceEvaluator]:
+        # Evidence will be sampled so we don't need to evaluate raw data.
+        return []
+
+    def init_evidence_types(
+            self, signal_models: List[SignalModel],
+            evidence_evaluators: List[EvidenceEvaluator]
+    ) -> List[EvidenceType]:
+        evidence_types = set(
+            [get_evidence_type(model) for model in self.signal_models])
+        return [EvidenceType.LM, *evidence_types]
+
+    def init_display(self) -> Display:
+        return NullDisplay()
+
+    def init_feedback(self) -> Optional[VisualFeedback]:
+        return None
+
+    def user_wants_to_continue(self) -> bool:
+        return True
+
+    def wait(self, seconds: Optional[float] = None) -> None:
+        """Override to do nothing"""
+
+    def present_inquiry(
+        self, inquiry_schedule: InquirySchedule
+    ) -> Tuple[List[Tuple[str, float]], bool]:
+        """Override ; returns empty timing info; always proceed for inquiry
+        preview"""
+        return [], True
+
+    def show_feedback(self, selection: str, correct: bool = True) -> None:
+        """Override to do nothing"""
+        # TODO: consider tracking metrics
+
+    def compute_button_press_evidence(
+            self, proceed: bool) -> Optional[Tuple[EvidenceType, List[float]]]:
+        # TODO: consider adding a sampler for this
+        return None
+
+    def compute_device_evidence(
+            self,
+            stim_times: List[List],
+            proceed: bool = True) -> List[Tuple[EvidenceType, List[float]]]:
+
+        current_state = self.get_sim_state()
+        self.logger.info("Computing evidence with sim_state:")
+        self.logger.info(current_state)
+
+        evidences = []
+
+        for model in self.signal_models:
+            sampler = self.samplers[model]
+            # This assumes that sampling is independent. Changes to the sampler API are needed if
+            # we need to provide the trial context of the last sample.
+            sampled_data = sampler.sample_data(current_state)
+            evidence = model.predict(sampled_data, self.current_symbols(),
+                                     self.alp)
+            evidence_type = model.metadata.evidence_type or EvidenceType.ERP
+            evidences.append((evidence_type, evidence))
+        return evidences
+
+    def exit_display(self) -> None:
+        """Close the UI and cleanup."""
+
+    def elapsed_seconds(self) -> float:
```

Comment:
Is it possible to estimate the just the computation time per symbol selection / next symbol querying? I think it would be more realistic if the formula here would be something like: `elapsed_time = inquiry_length + computation_time`

Reply:
The computation of time is surprisingly complicated. There is an estimate in the acquisition helper for how much time an inquiry will take (for acquisition buffering purposes), but there are some confounding factors. I will create a followup ticket because I want to give this more thought and perhaps provide a way for this to be configured.